atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/Transaction.java

```diff
@@ -37,41 +37,123 @@
 
 /**
  * Provides the methods for a transaction with the key-value store.
+ *
+ * In general: users may assume that if maps (including sorted maps) keyed on byte[] are returned to the user,
+ * they may be accessed, including via random access, via any byte array that is equivalent in terms of
+ * {@link java.util.Arrays#equals(byte[], byte[])}.
+ *
  * @see TransactionManager
  */
 public interface Transaction {
 
+    /**
+     * Returns a mapping of rows to {@link RowResult}s within {@code tableRef} for the specified {@code rows}, loading
```

Comment:
Meta comment: is this javadoc copied to the generated code? If not, this feels like nobody will actually read this and understand...

Reply:
There'll be a next part to this. These comments were written more for AtlasDB devs than for users (so we won't think it's acceptable to e.g. change the order of rows)